src/d/actor/d_a_tbox.cpp

```diff
@@ -451,23 +530,152 @@ s32 daTbox_c::getFuncType() {
 }
 
 /* 00000FF0-0000108C       .text checkNormal__8daTbox_cFv */
-void daTbox_c::checkNormal() {
-    /* Nonmatching */
+BOOL daTbox_c::checkNormal() {
+    s32 funcType = getFuncType();
+
+    if (funcType == FUNC_TYPE_NORMAL || funcType == FUNC_TYPE_EXTRA_SAVE_INFO || funcType == FUNC_TYPE_SWITCH_VISIBLE || funcType == FUNC_TYPE_GRAVITY) {
+        return TRUE;
+    }
+
+    if (mRoomNo == -1 || mRoomNo == 0x3F) {
+        return FALSE;
+    }
+
+    s32 swNo = getSwNo();
+    if (swNo >= 0xC0) {
+        return FALSE;
+    }
+
+    return dComIfGs_isSwitch(swNo, mRoomNo) ? TRUE : FALSE;
```

Comment:
for these tails, can you try out e.g.

```cpp
if (dComIfGs_isSwitch(sw, mRoomNo))
    return TRUE;

return FALSE;
```

or the inverse of that? curious to see if it works